### PR TITLE
Split out OutMessage and change "recording" to "capturing"

### PR DESCRIPTION
**Pull request by Bubba**
`OutMessage` now consists of two separate data types for messages that originate from the client and messages that originate from the server.

Recording is now called capturing and encodes message types based on these new types. It now saves the session to one file with timestamped messages and type information.

### CODE REVIEW

File: src/Language/Haskell/LSP/Control.hs [hunk 1]
```diff
@@ -68,41 +68,44 @@ runWithHandles hin hout dp h o recInFp recOutFp = do
   hSetEncoding  hout utf8
 
   -- Delete existing recordings if they exist
-  forM_ [recInFp, recOutFp] $ maybe (return ()) $ \f -> do
-    exists <- doesFileExist f
-    when exists $ removeFile f
+  maybe (return ()) removeFileIfExists captureFp
```
Comment (alanz) [inline src/Language/Haskell/LSP/Control.hs]: perhaps capture against a filename with a timestamp in it.  That way we do not lose something accidentally.

Eventually, that is.

**Comment by alanz** [comment]
Is there a matching change for hie?

**Comment by Bubba** [comment]
Yep, in https://github.com/Bubba/haskell-ide-engine/tree/capturing

File: src/Language/Haskell/LSP/Control.hs [hunk 2]
```diff
@@ -67,12 +67,12 @@ runWithHandles hin hout dp h o captureFp = do
   hSetBuffering hout NoBuffering
   hSetEncoding  hout utf8
 
-  -- Delete existing recordings if they exist
-  maybe (return ()) removeFileIfExists captureFp
-
+  timestamp <- formatTime defaultTimeLocale (iso8601DateFormat (Just "%H-%M-%S")) <$> getCurrentTime
```
Comment (alanz) [inline src/Language/Haskell/LSP/Control.hs]: Probably best to add yymmdd as well.

Comment (Bubba) [reply src/Language/Haskell/LSP/Control.hs]: [iso8601DateFormat](https://hackage.haskell.org/package/time-1.9.1/docs/Data-Time-Format.html#v:iso8601DateFormat) adds that for us

**Pull request merged**